### ISSUE TITLE
[Task] Refactor React.FunctionalComponent/React.FC using codemod-replace-react-fc-typescript

### ISSUE THREAD

**Issue by maxcao13** [cryostatio/cryostat-web#482]
> I haven't seen that before. I don't really have any thoughts on the matter or opinion one way or the other.
> 
> https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/
> 
> I tend to like the explicit/static types where possible, but sometimes it isn't really necessary - and if the type signature is actually lying to you, like it is about the implicit children, then it does more harm than good.
> 
> If you'd like to go through and refactor to remove all those type declarations later then I wouldn't be opposed.

Maybe I will do a refactor in the future ([the docs you linked has something for that anyways](https://github.com/gndelia/codemod-replace-react-fc-typescript)), but for now I think it may be best to stop using React.FC in `-web` in the future and stick with the usual 
```
const Foo = ( {prop1, prop2}: FooProps ): JSX.Element => {
   ...
   return ...
}
``` 
so that everything is using the same and preferred style. 

I'll open an issue for it, if that's okay with you anyways. This isn't a big deal by any means ^^. I'm not sure what is the best way to let everyone know though, haha.

_Originally posted by @maxcao13 in https://github.com/cryostatio/cryostat-web/issues/481#issuecomment-1183747147_

**Comment by maxcao13** [comment]
I will start this after https://github.com/cryostatio/cryostat-web/pull/478 is merged because of this comment on https://github.com/gndelia/codemod-replace-react-fc-typescript#codemod-replace-react-fc-typescript

> (It's recommended to run your favorite formatting tool after the codemod )

**Comment by maxcao13** [comment]
Reading this again after the recent merge and it seems like there's not really a need to refactor React.FC anymore since the implicit children problem is solved in React 18 https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-typescript-definitions. Is there any push back from changing this issue to updating cryostat-web from React 17.0.2 to React 18.*?

**Comment by andrewazores** [comment]
No, that sounds good, though it sounds like the upgrade may not be totally seamless judging by those release upgrade notes. We'll need to go through things carefully and make sure the changes don't cause any odd bugs. We're pretty close to the code freeze date for 2.2 release so this would need to be done quite soon to have any chance of making it in to the release.

**Comment by andrewazores** [comment]
https://github.com/openshift/console/blob/main/frontend/packages/console-dynamic-plugin-sdk/release-notes/4.22.md
